javascript/javascript3/week3/script.js

```diff
@@ -0,0 +1,285 @@
+//use class way original data
+let c = document.getElementById("myCanvas");
+c.width = window.innerWidth;
+c.height = window.innerHeight;
+let ctx = c.getContext("2d");
+class Circle {
+    constructor(x, y, r, startAngle, endAngle, fillColor) {
+        this.x = x,
+            this.y = y,
+            this.r = r,
+            this.startAngle = startAngle,
+            this.endAngle = endAngle,
+            this.fillColor = fillColor
+    }
+    draw() {
+
+        ctx.beginPath();
+        ctx.arc(
+            this.x,
+            this.y,
+            this.r,
+            this.startAngle,
+            this.endAngle,
+            this.fillColor);
+        ctx.stroke();
+        ctx.fillStyle = this.fillColor;
+        ctx.fill();
+    }
+}
+const timeInterval = setInterval(() => {
+    let allCircles = [];
+    let amount = 2;
+    let createCircle = function (circle) {
+        circle.draw();
+    };
+    for (let i = 0; i < amount; i++) {
+        let randomX = Math.random() * window.innerWidth;
+        let randomY = Math.random() * window.innerHeight;
+        let color =
+            "rgba(" +
+            Math.floor(Math.random() * 255) +
+            "," +
+            Math.floor(Math.random() * 255) +
+            "," +
+            Math.floor(Math.random() * 255) +
+            "," +
+            ".5" +
+            ")";
+
+        let c2 = new Circle(randomX, randomY, 20, 0, 2 * Math.PI, color);
+        allCircles.push(c2);
+        createCircle(allCircles[i]);
+    }
+}, 1000);
+
+
+//Follow the mouse - optional
+document.addEventListener('DOMContentLoaded', () => {
+    let mousePosX = 0,
+        mousePosY = 0,
+        mouseCircle = document.getElementById('mouse-circle');
+    document.onmousemove = (e) => {
+        mousePosX = e.pageX;
+        mousePosY = e.pageY;
+    }
+    let delay = 8,
+        revisedMousePosX = 0,
+        revisedMousePosY = 0;
+    function delayMouseFollow() {
+        requestAnimationFrame(delayMouseFollow);
+        revisedMousePosX += (mousePosX - revisedMousePosX) / delay;
+        revisedMousePosY += (mousePosY - revisedMousePosY) / delay;
+        mouseCircle.style.top = revisedMousePosY + 'px';
+        mouseCircle.style.left = revisedMousePosX + 'px';
+    }
+    delayMouseFollow();
+});
+
+//Getting into promises
+
+const renderInfo = document.querySelector(".promiseAll");
+const promiseAll = async function promiseAll() {
+    try {
+        const apiOne = "https://api.github.com/search/repositories?q=user:chunmeishui";
```

Comment:
use more meaningful name, like userChunmei

Reply:
oh yes, you are right. thanks